### PR TITLE
DisplaySettings: Use FileSystemAccessServer instead of FilePicker

### PR DESCRIPTION
**Pull request by huttongrabiel**
Use `FileSystemAccessServer` instead of `FilePicker` to select wallpaper in BackgroundSettings.

Fixes crash in DisplaySettings introduced in #17777

### CODE REVIEW

**Comment by huttongrabiel** [comment]
Did not know `FileSystemAccessServer` existed. Updated to use it. Thanks

File: Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp
Comment (LucasChollet) [inline Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp:10]: Is it really needed? I guess that it was indirectly included by `FilePicker.h`.

**Review by LucasChollet — COMMENTED** [review]
Small regression: we don't have the type filter anymore.
At least add a FIXME about that, but don't hesitate to add that functionality to FSAS.

Otherwise, LGTM!

File: Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp
Comment (huttongrabiel) [reply Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp:10]: Ya it is. It's an indirect include exactly as you said.

**Comment by huttongrabiel** [comment]
Sorry, what is the type filter regression that you are talking about?

**Comment by huttongrabiel** [comment]
*Note*: Don't merge this if #17904 gets merged first. One does not rely on the other but if the other gets merged first I'll update this to use that functionality.

**Comment by huttongrabiel** [comment]
Updated to use #17904

**Review by caoimhebyrne — APPROVED** [review]

File: Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp
Comment (caoimhebyrne) [inline Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp:105]: Does this path not get traversed when cancel is pressed?

**Review by caoimhebyrne — CHANGES_REQUESTED** [review]
Just one thing

File: Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp
Comment (LucasChollet) [reply Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp:105]: It does, good catch!
We should definitely silently return in this case.

Comment (caoimhebyrne) [reply Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp:105]: Most likely. All other usages of `FileSystemAccessClient::Client::the().open_file` that I can see just return early.
```cpp
if (response.is_error()
    return;
```

**Review by caoimhebyrne — APPROVED** [review]

**Review by ADKaster — APPROVED** [review]

**Pull request merged**
